developer-docs-site/docs/guides/resource-accounts.md

```diff
@@ -0,0 +1,28 @@
+---
+title: "Resource Accounts"
+slug: "resource-accounts"
+---
+
+# Resource Accounts
+
+A [resource account](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/resource_account.move) is a developer feature used to manage resources independent of an account managed by a user, specifically publishing modules and automatically signing for transactions. For example, a developer may use a resource account to manage an account for module publishing, say managing a contract. The contract itself does not require a signer post initialization. A resource account gives you the means for the module to provide a signer to other modules and sign transactions on behalf of the module.
+
+## Restrictions
+
+In Aptos, a resource account is created based upon the SHA3-256 hash of the source's address and additional seed data. A resource account can be created only once. An entity may call `create_account` in an attempt to claim an account ahead of the creation of a resource account. But if a resource account is found, Aptos will transition ownership of the account over to the resource account. This is done by validating that the account has yet to execute any transactions and that the `Account::signer_capbility_offer::for` is none. The probability of a collision where someone has legitimately produced a private key that maps to a resource account address is improbably low.
+
+## Setup
+
+To properly set a resource account up, you will need to [initialize](https://github.com/aptos-labs/aptos-core/blob/2e9d8ee759fcd3f6e831034f05c1656b1c48efc4/aptos-move/move-examples/mint_nft/sources/minting.move#L73) the `mint_nft` module and retrieve the signer capability from both the resource account and module account. To do so, call `create_resource_account_and_publish_package` to publish the module under the resource account's address.
```

Comment:
Just want to call out that there's an issue with the create_resource_account_and_publish_package flow currently where if someone uses it to deploy a new module, they would completely lose control over the resource account and have no way to retrieve the signer/signercap. cc @chloeqjz. My suggestion would be to omit this from documentation until we fix the issue.

Reply:
cc @davidiw

Reply:
The alternative flow is:
1. Create the resource account via resource_account::create_resource_account to make sure the original account still has control over the resource account.
2. Publish the module to the resource account using the standard module publishing flow using the original account's auth key.
3. Once all initializations are done, developer can optionally remove the auth key for the resource account via retrieve_resource_account_cap to ensure the resource account can now only be controlled via SignerCapability. This has to be done in a script as retrieve_resource_account_cap returns the SignerCapability.

Reply:
Thanks, @movekevin . Should these steps replace the introductory paragraph and steps I drafted in toto, or only  the intro paragraph?

Reply:
@movekevin I talked with David about this and created a [PR](https://github.com/aptos-labs/aptos-core/pull/5388) to fix the low. 

so the previous flow is that the NFT collection and module data were both controlled by the `resource signer`. In the new flow, the NFT collection is controlled by the `resource signer` (to mint tokens programmatically) and the `ModuleData` is controlled by the `admin` - the `admin` can update time stamp, update if the minting is enabled, and update the pubic key which the signature will validate against.

Reply:
@clay-aptos thanks for your patience when we're working thru this!

I think it might be a good idea to mention that once they use `create_resource_account_and_publish_package`, they will no longer have access to their resource account (and this is by design, because people use resource account when they want the contract to be autonomous & immutable). If the user use `create_resource_account` or `create_resource_account_and_fund`, they'd still have access to the resource account's signer, until they explicitly give it up by calling `retrieve_resource_account_cap`. 

We also updated the NFT tutorial [here](https://github.com/aptos-labs/aptos-core/pull/5388) to make it less confusing! Lmk if you have any questions!!

Reply:
> @clay-aptos thanks for your patience when we're working thru this!
> 
> I think it might be a good idea to mention that once they use `create_resource_account_and_publish_package`, they will no longer have access to their resource account (and this is by design, because people use resource account when they want the contract to be autonomous & immutable). If the user use `create_resource_account` or `create_resource_account_and_fund`, they'd still have access to the resource account's signer, until they explicitly give it up by calling `retrieve_resource_account_cap`.
> 
> We also updated the NFT tutorial [here](https://github.com/aptos-labs/aptos-core/pull/5388) to make it less confusing! Lmk if you have any questions!!

Hi Chloe, this sounds like we should re-evaluate the entire set of resource account setup steps (and links!) I've compiled. Still, I have presented all three options, explaining their uses, like so:

You have three options for creating a resource account:
* `create_resource_account_and_publish_package` - creates the resource account and results in loss of access to the resource account by design, because resource accounts are used to make contracts autonomous and immutable.
* `create_resource_account_and_fund` - creates the resource account and funds it, retaining access to the resource account's signer until explicitly calling `retrieve_resource_account_cap`.
* `create_resource_account` - merely creates the resource account but doesn't fund it, retaining access to the resource account's signer until explicitly calling `retrieve_resource_account_cap`.

Want to review the remaining steps for needed changes based upon your PR? Or do so in a brief meeting with me? Thanks!

Reply:
That's a really summary of the 3 methods. LGTM.

Reply:
yeah that sounds good! for the remaining steps based upon my PR, let's go over it in a meeting! send me an invite whenever it's convenient for you : )